File: Boats to Save People.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        hev = len(people)-1
        lig = 0
        boat = 0

        while hev >= lig:
            if people[lig] + people[hev] <= limit:
                boat += 1
                lig += 1 
                hev -= 1
            else:
                boat += 1
                hev -= 1
        return boat

File: Boats to Save People.js
var numRescueBoats = function(people, limit) {
    people.sort((a,b)=>a-b);
    let left=0, right=people.length-1 , boat=0;

    while (left<=right){
        if (people[left]+people[right]<=limit){
            left ++;
            right --;
            boat ++;
        }
        else{
            right --;
            boat ++;
        }
    }
    return boat ; 
};